x-pack/test/alerting_api_integration/observability/threshold_rule/avg_pct_fired.ts

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { cleanup, generate } from '@kbn/infra-forge';
+import { Aggregators, Comparator } from '@kbn/observability-plugin/common/threshold_rule/types';
+import { FIRED_ACTIONS_ID } from '@kbn/observability-plugin/server/lib/rules/threshold/threshold_executor';
+import expect from '@kbn/expect';
+import { OBSERVABILITY_THRESHOLD_RULE_TYPE_ID } from '@kbn/observability-plugin/common/constants';
+import { createIndexConnector, createRule } from '../helpers/alerting_api_helper';
+import { createDataView, deleteDataView } from '../helpers/data_view';
+import { waitForAlertInIndex, waitForRuleStatus } from '../helpers/alerting_wait_for_helpers';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+
+// eslint-disable-next-line import/no-default-export
+export default function ({ getService }: FtrProviderContext) {
+  const esClient = getService('es');
+  const supertest = getService('supertest');
+  const esDeleteAllIndices = getService('esDeleteAllIndices');
+  const logger = getService('log');
+
+  describe('Threshold rule - AVG - PCT - FIRED', () => {
```

Comment:
What are the differences between `Threshold rule - AVG - PCT - FIRED` and `Threshold rule - AVG - US - FIRED`?
Initially, I was thinking about having multiple scenarios to check the reason to make sure they are properly formatted, I am curious to know what other differences we can test. 

FYI, I created another [ticket](https://github.com/elastic/kibana/issues/162950) for the action variables, then we can also check the reason to ensure the correct formating is used.

Reply:
> What are the differences between Threshold rule - AVG - PCT - FIRED and Threshold rule - AVG - US - FIRED?

Field data type % and microseconds 

> Initially, I was thinking about having multiple scenarios to check the reason to make sure they are properly formatted, I am curious to know what other differences we can test.

I am not sure If understand what you mean, but please create a ticket with the check you want to do with the scenarios you have in mind.

Reply:
My question is what do we test for different field data types related to their type? Can you please point me to the code that is different between these two tests related to their types?

Reply:
@maryam-saeidi, I'm fulfilling the 1st scenario you requested in your [issue](https://github.com/elastic/kibana/issues/161237):
> Multiple conditions with different types of fields (such as number, string, percent) without grouping (Can we have one condition for every condition type? 😄)

I thought you knew what is different, so you asked to cover it.

Reply:
Changing the field type will make many areas of code to be tested. I will list some files to answer the question 
`x-pack/plugins/observability/server/lib/rules/threshold/lib/get_data.ts`
`x-pack/plugins/observability/server/lib/rules/threshold/lib/metric_query.ts`
`x-pack/plugins/observability/server/lib/rules/threshold/threshold_executor.ts`
`x-pack/plugins/observability/server/lib/rules/threshold/lib/create_percentile_aggregation.ts`
... and maybe others

As this is an integration test (not a unit test), asking about what lines of code are covered is inadequate and hard to answer.

Reply:
> I thought you knew the difference, so you asked to cover it.

Yes, the difference was the `reason` message that we are not checking at the moment.

Reply:
> As this is an integration test (not a unit test), asking about what lines of code are covered is inadequate and hard to answer.

By the line of code, I meant related to the new tests that you added, not the code behind the scene, something maybe like: the value that is saved in AAD is formatted differently for different field types.